compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3380,6 +3383,8 @@ public void visit(BLangPanic panicNode) {
     @Override
     public void visit(BLangXMLNSStatement xmlnsStmtNode) {
         xmlnsStmtNode.xmlnsDecl = rewrite(xmlnsStmtNode.xmlnsDecl, env);
+        //need to propagate the local namespace to be used at the select clause
+        this.stmtsToBePropagatedToQuery.add(xmlnsStmtNode);
```

Comment:
Here we are just blindly adding xmlns declaration without taking the scope into account. Therefore, namespaces declared outside the scope also get declared in the query.  If shadowing takes place could this be problem?

e.g.
```bal
xmlns "x.org" as localNS;

function foo() {
    xmlns "y.org" as localNS;
}

function bar() {
    xmlns "z.org" as localNS;
    foo();
    xml x = from int i in 1 ... 3
        select xml `<empRecord employeeId="${i}"><localNS:id>${i}</localNS:id></empRecord>`;
}
```

Here `localNS` in query refers to whatever last visited `BLangXMLNSStatement` I guess. But, visiting order should not be matter. Scope should matter.